pkg/workflows/compute_test.go

```diff
@@ -0,0 +1,204 @@
+package workflows_test
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/consensus/ocr3"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/targets/chainwriter"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/triggers/streams"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/testdata/fixtures/capabilities/notstreams"
+)
+
+func TestCompute(t *testing.T) {
+	nsf, err := values.CreateMapFromStruct(map[string]any{"Arg0": notstreams.Feed{
+		Price: notstreams.FeedPrice{
+			PriceA: "12.3",
+			PriceB: "32.1",
+		},
+		Timestamp:     123,
+		FullReport:    "report",
+		ReportContext: "context",
+		Signatures:    []string{"sig1", "sig2"},
+	}})
+	require.NoError(t, err)
+
+	t.Run("creates correct workflow spec", func(t *testing.T) {
+		workflow := createWorkflow(convertFeed)
+
+		spec, err2 := workflow.Spec()
+		require.NoError(t, err2)
+		expectedSpec := workflows.WorkflowSpec{
+			Name:  "name",
+			Owner: "owner",
+			Triggers: []workflows.StepDefinition{
+				{
+					ID:             "notstreams@1.0.0",
+					Ref:            "trigger",
+					Inputs:         workflows.StepInputs{},
+					Config:         map[string]any{"maxFrequencyMs": 5000},
+					CapabilityType: capabilities.CapabilityTypeTrigger,
+				},
+			},
+			Actions: []workflows.StepDefinition{
+				{
+					ID:  "internal!!custom_compute@1.0.0",
```

Comment:
Making this more visible, to keystone team members, what's a good name to take that won't be in the real registry?  Is the prefix ok to use?

Reply:
Maybe `__internal__`?